packages/react-core/src/next/components/Wizard/__tests__/Wizard.test.tsx

```diff
@@ -301,4 +301,44 @@ describe('Wizard', () => {
     await user.click(screen.getByRole('button', { name: 'Cancel' }));
     expect(onClose).toHaveBeenCalled();
   });
+
+  it('unmounts inactive steps by default', async () => {
+    const user = userEvent.setup();
+
+    render(
+      <Wizard>
+        <WizardStep id="step-1" name="Test step 1">
+          Step 1 content
+        </WizardStep>
+        <WizardStep id="step-2" name="Test step 2">
+          Step 2 content
+        </WizardStep>
+      </Wizard>
+    );
+
+    await user.click(screen.getByRole('button', { name: 'Test step 2' }));
+
+    expect(screen.queryByText('Step 1 content')).toBeNull();
+    expect(screen.getByText('Step 2 content')).toBeVisible();
+  });
+
+  it('keeps inactive steps mounted when unmountInactiveSteps is enabled', async () => {
+    const user = userEvent.setup();
+
+    render(
+      <Wizard unmountInactiveSteps={false}>
+        <WizardStep id="step-1" name="Test step 1">
+          Step 1 content
+        </WizardStep>
+        <WizardStep id="step-2" name="Test step 2">
+          Step 2 content
+        </WizardStep>
+      </Wizard>
+    );
+
+    await user.click(screen.getByRole('button', { name: 'Test step 2' }));
+
+    expect(screen.getByText('Step 1 content')).toBeInTheDocument();
+    expect(screen.getByText('Step 2 content')).toBeVisible();
+  });
```

Comment:
I think it's useful to know that passing the prop to `Wizard` leads to this user interaction. Especially since this takes into account the usage of the default navigation that comes with Wizard. I've added some tests to `WizardToggle` though that add some expectations for `unmountInactiveSteps` as well.